hazelcast/src/test/java/classloading/AbstractThreadLeakTest.java

```diff
@@ -35,7 +36,11 @@ public final void getOldThreads() {
     }
 
     @After
-    public final void assertThreadLeaks() {
-        assertHazelcastThreadShutdown(oldThreads);
+    public void assertThreadLeaks() {
+        try {
+            assertHazelcastThreadShutdown(oldThreads);
+        } finally {
```

Comment:
I don't believe shutting down should be a responsibility of an assert method. If you need to cleanup something, this should be part of an 'after' method.

Reply:
Yes, this was a different approach to test design compared to the other tests. Unfortunately, this was the way it was designed originally when the thread leak tests were written for member and client. I just added new tests to it. Should we refactor it in this PR or should it be a separate PR for refactoring?